pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -262,27 +262,14 @@ func convertVirtualService(r ConfigContext) []config.Config {
 }
 
 func buildHTTPVirtualServices(
-	refs AllowedReferences,
+	ctx ConfigContext,
 	obj config.Config,
-	gateways map[parentKey]map[k8s.SectionName]*parentInfo,
-	domain string,
 	gatewayRoutes map[string]map[string]*config.Config,
 	meshRoutes map[string]map[string]*config.Config,
 ) {
 	route := obj.Spec.(*k8s.HTTPRouteSpec)
-	var splitRules []k8s.HTTPRouteRule
-	for _, r := range route.Rules {
-		if len(r.Matches) > 1 {
-			// split the rule to make sure each rule has up to one match
-			matches := r.Matches
-			for _, m := range matches {
-				r.Matches = []k8s.HTTPRouteMatch{m}
-				splitRules = append(splitRules, r)
-			}
-		}
-	}
```

Comment:
This change is for https://github.com/istio/istio/pull/39085#discussion_r906141636